gobblin-temporal/src/main/java/org/apache/gobblin/temporal/yarn/YarnService.java

```diff
@@ -755,6 +776,18 @@ private ImmutableMap.Builder<String, String> buildContainerStatusEventMetadata(C
     return eventMetadataBuilder;
   }
 
+  /**
+   * Generates a unique allocation request ID for the given worker profile and store the id to profile mapping.
+   *
+   * @param workerProfile the worker profile for which the allocation request ID is generated
+   * @return the generated allocation request ID
+   */
+  protected long generateAllocationRequestId(WorkerProfile workerProfile) {
```

Comment:
naming-wise, the side-effect of putting into the map seems critical for maintainers to notice. maybe `generateAndStoreByUniqueAllocationRequestId` or simply `storeByUniqueAllocationRequestId`?

Reply:
using `storeByUniqueAllocationRequestId`